openshift-install-powervs

```diff
@@ -1281,36 +1283,36 @@ function process_varfile {
   done
 }
 
-function display_version {
-  echo "Version $VERSION"
-  exit 0
-}
-
-function main {
-  mkdir -p ./logs
-  vars=""
-
-  # Only use sudo if not running as root
-  [ "$(id -u)" -ne 0 ] && SUDO=sudo || SUDO=""
-
+#-------------------------------------------------------------------------
+# Perform platform checks
+#-------------------------------------------------------------------------
+function platform_checks {
   if [[ $(uname -m) != "x86_64" ]]; then
     warn "Only 64-bit x86 machines are supported"
     error "Unsupported machine: $(uname -m)" 1
   fi
   PLATFORM=$(uname)
   case "$PLATFORM" in
     "Darwin")
+      OS_VERSION=$(sw_vers -productVersion 2>/dev/null || echo "")
+      if [[ $IGNORE_OS -eq 0 ]] && [[ "$OS_VERSION" != "10.15"* && "$OS_VERSION" != "11."* ]]; then
+        error "Unsupported os version-$OS_VERSION of Mac, please use '-ignore-os-checks' option if you still want to run it."
+      fi
       PACKAGE_MANAGER="brew"
       OS="darwin"
       IBM_CLI_OS="osx"
       OC_CLI_OS="mac"
       ARCHIVE_TYPE="tar.gz"
       ;;
     "Linux")
-      # Linux distro, e.g "Ubuntu", "RedHatEnterpriseWorkstation", "RedHatEnterpriseServer", "CentOS", "Debian"
+      # Linux distro, e.g "Ubuntu", "RedHatEnterpriseWorkstation", "RedHatEnterpriseServer", "CentOS"
       DISTRO=$(lsb_release -ds 2>/dev/null || cat /etc/*release 2>/dev/null | head -n1 || uname -om || echo "")
-      if [[ "$DISTRO" != *Ubuntu* &&  "$DISTRO" != *Red*Hat* && "$DISTRO" != *CentOS* && "$DISTRO" != *Debian* && "$DISTRO" != *RHEL* && "$DISTRO" != *Fedora* ]]; then
-        warn "Linux has only been tested on Ubuntu, RedHat, CentOS, Debian and Fedora distributions please let us know if you use this utility on other distributions"
+      OS_VERSION=$(lsb_release -rs 2>/dev/null || cat /etc/*release 2>/dev/null | grep "VERSION_ID" | awk -F "=" '{print $2}' | sed 's/"*//g' || echo "")
+      if [[ $IGNORE_OS -eq 0 ]] && [[ "$DISTRO" != *Ubuntu* && "$DISTRO" != *Red*Hat* && "$DISTRO" != *CentOS* ]]; then
+        error "Linux has only been tested on Ubuntu, RedHat, and CentOS distributions please use '-ignore-os-checks' option to run this utility on other distributions."
+      fi
+      if [[ $IGNORE_OS -eq 0 ]] && [[ "$OS_VERSION" != "16.04" && "$OS_VERSION" != "18.04" && "$OS_VERSION" != "20.04" && "$OS_VERSION" != "8"* ]]; then
+        error "Unsupported os version-$OS_VERSION of $DISTRO, please use '-ignore-os-checks' option if you still want to run it."
```

Comment:
"Unsupported OS version ($OS_VERSION) for $DISTRO"

Reply:
Done